aggregations/db_tables/deployed_contracts.py

```diff
@@ -60,3 +81,102 @@ def start_of_range(self, timestamp: int) -> int:
+                    print("Retrying fetching contract code...")
+                    traceback.print_exc()
+                except Exception:
+                    print("Retrying fetching contract code...")
+                    traceback.print_exc()
+                else:
+                    break
+            return base64.b64decode(response["code_base64"])
+
+        sql_missing_sdk_type = """
+            SELECT contract_code_sha256, deployed_to_account_id, deployed_at_block_hash
+            FROM deployed_contracts
+            WHERE contract_sdk_type = '' AND deployed_at_block_hash != ''
+            LIMIT 10
+        """
+
+        with self.analytics_connection.cursor() as analytics_cursor:
+            while True:
+                analytics_cursor.execute(sql_missing_sdk_type)
+                unknown_contracts = analytics_cursor.fetchall()
+                if not unknown_contracts:
+                    break
+                print(unknown_contracts)
+
+                contract_sdk_types = []
+                for (
+                    contract_code_sha256,
+                    contract_account_id,
+                    deployed_at_block_hash,
+                ) in unknown_contracts:
+                    print("fetching ", contract_account_id)
+                    contract_code = view_code(
+                        contract_account_id, deployed_at_block_hash
+                    )
+
+                    if contract_code == b"":
+                        contract_sdk_type = "EMPTY"
+                    elif not contract_code.startswith(b"\0asm"):
+                        contract_sdk_type = "NOT_WASM"
+                    else:
+                        likely_sdk_types = set()
+                        if (
+                            b"__data_end" in contract_code
+                            and b"__heap_base" in contract_code
+                        ):
+                            likely_sdk_types.add("RS")
+                        if (
+                            b"JS_TAG_MODULE" in contract_code
+                            and b"quickjs-libc-min." in contract_code
+                        ):
+                            likely_sdk_types.add("JS")
+                        if (
+                            b"l\x00i\x00b\x00/\x00a\x00s\x00s\x00e\x00m\x00b\x00l\x00y\x00s\x00c\x00r\x00i\x00p\x00t"
+                            in contract_code
+                            or b"~lib/near-sdk-core/collections/persistentMap/PersistentMap"
+                            in contract_code
+                        ):
+                            likely_sdk_types.add("AS")
+
+                        # Only set the sdk type if exactly one match is received since if we matched multiple, it is impossible to make a call.
+                        contract_sdk_type = (
+                            likely_sdk_types.pop()
+                            if len(likely_sdk_types) == 1
+                            else "UNKNOWN"
```

Comment:
Let's put this if-else code to the function. It's much easier to read more flattened code, and this piece has a separate meaning

Reply:
Good point. Before I formatted it with black, it was more readable

Reply:
You haven't extracted that